elrond-wasm/src/storage/mappers/map_storage_mapper.rs

```diff
@@ -227,3 +235,155 @@ where
+	K: TopEncode + TopDecode + Clone + 'static,
+	V: StorageMapper<SA> + StorageClearable,
+{
+	/// Ensures a value is in the entry by inserting the default if empty, and returns
+	/// an `OccupiedEntry`.
+	pub fn or_insert_default(self) -> OccupiedEntry<'a, SA, K, V> {
+		match self {
+			Entry::Occupied(entry) => entry,
+			Entry::Vacant(entry) => entry.insert_default(),
+		}
+	}
+
+	/// Returns a reference to this entry's key.
+	pub fn key(&self) -> &K {
+		match *self {
+			Entry::Occupied(ref entry) => entry.key(),
+			Entry::Vacant(ref entry) => entry.key(),
+		}
+	}
+
+	/// Provides in-place mutable access to an occupied entry before any
+	/// potential inserts into the map.
+	pub fn and_modify<F>(self, f: F) -> Self
+	where
+		F: FnOnce(&mut V),
+	{
+		match self {
+			Entry::Occupied(mut entry) => {
+				entry.update(f);
+				Entry::Occupied(entry)
+			},
+			Entry::Vacant(entry) => Entry::Vacant(entry),
+		}
+	}
+}
+
+impl<'a, SA, K, V> Entry<'a, SA, K, V>
+where
+	SA: StorageReadApi + StorageWriteApi + ErrorApi + Clone + 'static,
+	K: TopEncode + TopDecode + Clone + 'static,
+	V: StorageMapper<SA> + StorageClearable,
+{
+	/// Ensures a value is in the entry by inserting the default value if empty,
+	/// and returns an `OccupiedEntry`.
+	pub fn or_default(self) -> OccupiedEntry<'a, SA, K, V> {
+		match self {
+			Entry::Occupied(entry) => entry,
+			Entry::Vacant(entry) => entry.insert_default(),
+		}
+	}
+}
+
+impl<'a, SA, K, V> VacantEntry<'a, SA, K, V>
+where
+	SA: StorageReadApi + StorageWriteApi + ErrorApi + Clone + 'static,
+	K: TopEncode + TopDecode + Clone + 'static,
+	V: StorageMapper<SA> + StorageClearable,
+{
+	/// Gets a reference to the key that would be used when inserting a value
+	/// through the VacantEntry.
+	pub fn key(&self) -> &K {
+		&self.key
+	}
+
+	/// Sets the value of the entry with the `VacantEntry`'s key,
+	/// and returns an `OccupiedEntry`.
+	pub fn insert_default(self) -> OccupiedEntry<'a, SA, K, V> {
+		self.map.insert_default(self.key.clone());
+		OccupiedEntry {
+			key: self.key,
+			map: self.map,
+			_marker: PhantomData,
+		}
+	}
+}
+
+impl<'a, SA, K, V> OccupiedEntry<'a, SA, K, V>
+where
+	SA: StorageReadApi + StorageWriteApi + ErrorApi + Clone + 'static,
+	K: TopEncode + TopDecode + Clone + 'static,
+	V: StorageMapper<SA> + StorageClearable,
+{
+	/// Gets a reference to the key in the entry.
+	pub fn key(&self) -> &K {
+		&self.key
+	}
+
+	/// Gets the value in the entry.
+	pub fn get(&self) -> V {
+		self.map.get(&self.key).unwrap()
+	}
+
+	/// Syntactic sugar, to more compactly express a get, update and set in one line.
+	/// Takes whatever lies in storage, apples the given closure and saves the final value back to storage.
+	/// Propagates the return value of the given function.
+	pub fn update<R, F: FnOnce(&mut V) -> R>(&mut self, f: F) -> R {
+		let mut value = self.get();
+		f(&mut value)
+	}
+
+	/// Takes the value of the entry out of the map, removing it.
```

Comment:
Comment sounds a bit weird. Maybe rephrase to "Removes the entry from the map" or similar

Reply:
Done